### ISSUE TITLE
[Suggestion] Website to browse themes online

### ISSUE THREAD

**Issue by itsnotrin** [archcraft-os/archcraft-theme-manager#4]
We could theoretically create a dynamic site using something like EJS that just renders each of the themes into a little maybe boostrap card and creates cards automatically after polling them from this repo, some users would prefer to browse the site ahead of time or just not install an app.

Happy to draft up a site in paint or some basic code or even code the whole thing if you like the idea?

**Comment by itsnotrin** [comment]
Along with this we could potentially add automated submissions online as well as add other information to themes.json such as description of the theme, recommended packages perhaps too?

**Comment by benjibrown** [comment]
Potentially yes, however I feel it should not be priority and that the current theme manager should be prioritized.

**Comment by itsnotrin** [comment]
Could work alongside it perhaps as their development isn't particularly tied into each other if it's just pulling info from a git repo

**Comment by T-Dynamos** [comment]
> some users would prefer to browse the site ahead of time or just not install an app.

App is exclusively made for Archcraft, It may be pre-installed in next release.

> Along with this we could potentially add automated submissions online 

This may not be appropriate, beacause we need to verify each info which user submits ( including verify the tar file).

> add other information to themes.json such as description of the theme, recommended packages perhaps too?

We could also do it.

If you can start it, You could. And if you have any ui design examples, give them. I will try implementing.

**Comment by itsnotrin** [comment]
> App is exclusively made for Archcraft, It may be pre-installed in next release.

It's not just useful for Archcraft users, people who are looking into if they want to get Archcraft could view it or anyone in general could view it on the go


> This may not be appropriate, beacause we need to verify each info which user submits ( including verify the tar file).

I agree, either there needs to be some sort of system to automatically check it or we manually verify


> If you can start it, You could. And if you have any ui design examples, give them. I will try implementing.

For sure, I'll make my own repo in a bit and if you guys like the idea I'll happily transfer ownership of it

**Comment by benjibrown** [comment]
You'd have to de compress the archive and check for valid files (install.sh, preview.png etc) if you wanted to do it automatically. But that risks someone putting something malicious into the archive. Really, we should just manually check and also give an option for the user to download the source of a theme if they would like to verify its trustworthiness/what it does behind the scenes.

**Comment by benjibrown** [comment]
You definitely don't want the server to decompress user uploaded archives automatically when uploaded

**Comment by itsnotrin** [comment]
Agreed, maybe later on, there can be some automated verification with a VM or something, but for the time being, that's out of the scope of this project
